### ISSUE TITLE
XenServer: Error mounting bootstrap ISO

### ISSUE THREAD

**Issue by jefflill** [nforgeio/neonKUBE#1495]
Single node (with cStor) cluster provisioning is failing sometimes on XenServer with:

```
 [System.InvalidOperationException]: Sequence contains no elements
```

This happens when we're trying to mount the bootstrap ISO to the node's VM.  This is the second time I've seen this the past week.  We don't get a stack trace for for some reason.  Here's what we see in the **cluster.log:**

```
===============================================================================
STEP 6: provision virtual machine(s)


FAULTED HOSTS:

XCP-HOME:   *** FAULT: [System.InvalidOperationException]: Sequence contains no elements

# CLUSTER-END-PREPARE-FAILED ##################################################
```

**Comment by jefflill** [comment]
This must be happening in `XenServerHostingManager.ProvisionVM()` here:

```
// Extend the primary partition and file system to fill 
// the virtual disk.  Note that we're not going to do
// this if the specified disk size is less than or equal
// to the node template's disk size (because that
// would fail).
//
// Note that there should only be one unpartitioned disk
// at this point: the OS disk.

var partitionedDisks = node.ListPartitionedDisks();
var osDisk           = partitionedDisks.Single();   // <--- empty list
```

**Comment by jefflill** [comment]
BREAKING: I've modified:

```
ILinuxSshProxy.ListDisks(bool includeFloppy = false) --> ListDisks(bool fixedDiskOnly = true)
```

This method now checks the **rm** (removable) property returned for disks returned by `lsblk` and ignores removable disks by default.  This results in ignoring mounted ISO disks as an other removable disks such as USB drives.  This has close to the same behavior when no parameter is passed but will break existing apps that actually pass a value.

We never pass a parameter at this time, so we'll go with this change.

**Comment by jefflill** [comment]
**FIXED**

**Comment by jefflill** [comment]
This one has been bothering us very intermittently for over a year, and I think I've finally figured this out while debugging the Ubuntu 22.04 upgrade.

The **neon-init** service script assumes that the DVD is already mounted and available because we actually mount it before starting the VM.  The problem is that although the DVD is _physically mounted_ it may not have been detected by the OS at the time **neon-init** runs.

I've modified **neon-init** to retry DVD detection for 30 seconds with a 5 second retry interval.  This should correct this once and for all!
